docs_src/design/adr/security/0009-Secure-Bootstrapping.md

```diff
@@ -0,0 +1,453 @@
+These services are:
+
+* **Consul**: Requires TLS certificate set by configuration file or command line, with a TLS certificate injected into the container.
+
+* **Vault**: As Vault's database is encrypted, Vault cannot natively bootstrap its own TLS certificate.  Requires TLS certificate to be injected into container and its location set in a configuration file.
+
+* **PostgreSQL**: Requires TLS certificate to be injected into '$PGDATA'.
+
+* **Kong (admin)**: Requires environment variable to be set to secure admin port with TLS, with a TLS certificates injected into the container.
+
+* **Kong (external)**: Requires a bring-your-own (BYO) external certificate,
+or as a fallback, a default one should be generated using a configurable external hostname.
+
+* **Redis (v6)**: Requires TLS certificate set by configuration file or command line, with a TLS certificate injected into the container.
+
+* **Mosquitto**: Requires TLS certificate set by configuration file, with a TLS certificate injected into the container.
+
+Additionally, every EdgeX *microservice consumer* will require access to the root CA
+for certificate verification purposes,
+and every EdgeX *microservice server* will need a TLS leaf certificate and private key.
+
+There are no current plans for mutual auth TLS.
+
+#### Database passwords
+
+There are two databases currently used in EdgeX Hanoi:
+
+* **PostgreSQL** is the storage backend for Kong.
+  EdgeX provisions a the PostgeSQL container with a superuser
+  account and password, and provides the same to Kong.
+  This password can be a derived secret or random secret.
+
+* **Redis**: is the general-purpose database for EdgeX Hanoi.
+  Redis supports only a single password.
+  This password should be provisioned into the secret store
+  before services start running in production mode.
+  This password can be a derived secret or random secret.
+
+#### Kong admin credential
+
+The Kong admin port can be secured via
+[Kong API Loopback](https://docs.konghq.com/2.0.x/secure-admin-api/#introduction).
+After initial configuration of Kong,
+an authentication credential can be generated (derived or random)
+and stored in the secret store,
+and an authtnication plugin such as the HTTP Basic authentication plugin
+can be applied to the route.
+
+### Summary of bootstrapping secrets
+
+In summary, the bootstrapping secrets are:
+
+* Secrets seed (supplied externally to the framework in some hardened configurations)
+* Consul Serf encryption key (used to control cluster membership)
+* Consul management token (used to configure Vault)
+* Consul ACL token (used to configure Vault if Consul is used as Vault's backing store)
+* Vault master key (used to encrypt Vault's secrets)
+* TLS assets for a **local certificate authority**,
+  and **intermediate certificate authority**,
+  Consul,
+  Vault,
+  PostgreSQL,
+  Kong (admin),
+  Kong (external),
+  Redis (v6),
+  Mosquitto
+* PostgreSQL superuser password
```

Comment:
FWIW this is being worked on internally at Kong such that the migration credentials (privileged) can be separate from the admin credentials (crud) can be separated from proxy credentials (read/update with some insert exceptions).

I'm not able to provide a timeline but even so should continue down this route 👍

Reply:
Closing this comment as this section has been removed from the ADR.